overlord/aspectstate/aspectstate.go

```diff
@@ -52,7 +52,7 @@ func Set(st *state.State, account, bundleName, aspect, field string, value inter
 
 	asp := aspectBundle.Aspect(aspect)
 	if asp == nil {
-		return notFound("aspect %q was not found", aspect)
+		return notFound("aspect %s/%s/%s was not found", account, bundleName, aspect)
```

Comment:
wouldn't it make sense for notFound to take account, bundleName and aspect but not the formatting string? it is called the same all the time atm

Reply:
also fwiw similar errors like asserts.NotFoundError and config.NoOptionError are more structured

Reply:
We used aspects.NotFoundError for other cases as well (e.g., an aspect was found but the Get can't find a field), which is why the error was very loose. I split it into two more specific errors, AspectNotFoundError and ValueNotFoundError, so that the first can be more structured.
I made some general consistency and handling improvements to the errors. One thing that may look odd is that I removed the not found error that was being returned from the Mock function because it doesn't make sense for that mocking function to do any checks and, since it was, the actual error handling path was being covered by the tests.